htdocs/landing/index.php

```diff
@@ -21,7 +21,27 @@ <h1 class="Landing_Welcome">Welcome to GOCDB</h1>
             <p>Use of GOCDB is governed by the <a class="docLink hover" href="aup.html">EGI Acceptable Use Policy</a> which places restrictions on your use of the service.</p>
             <p>The <a class="docLink hover" href="privacy.html">GOCDB Privacy Notice</a> describes what personal data is collected and why, and your rights regarding this data.</p>
             <p> Please read these documents before accessing GOCDB.</p>
-            <a href="/portal/" class="button">Access GOCDB</a>
+          </div>
+          <div style="width: 80%; margin-left: auto; margin-right: auto;">
+            <a href="/portal/" style="width:73%; font-size:1.7em" class="button">Access GOCDB via X.509 Certificate</a>
+            <p>or</p>
+            <p>Access GOCDB using one of the following:</p>
+            <div>
+                <?php
+                $hostname = $_SERVER['HTTP_HOST'];
+                $egi_target = urlencode("https://" . $hostname . "/portal/");
+                $egi_redirect = "https://" . $hostname . "/Shibboleth.sso/Login?target=" . $egi_target;
+                if($_SERVER['REQUEST_URI'] === "/"){
+                    $iam_target = "target_link_uri=" . urlencode("https://" . $hostname . "/portal/");
+                }
+                else{
+                    $iam_target=ltrim($_SERVER['REQUEST_URI'], '/?');
+                }
+                $iam_redirect = "https://" . $hostname . "/portal/redirect_uri?iss=https%3A%2F%2Firis-iam.stfc.ac.uk%2F&" . $iam_target;
```

Comment:
Why is it not possible to use urlencode() here?

Reply:
Do you mean on the iam_target or iam_redirect?

Reply:
iam_redirect - urlencode("https://iris-iam.stfc.ac.uk/") or similar ?

Reply:
That bit is I think already urlencoded and just pasted in

Reply:
Granted. It's really a style/readability comment. You're welcome to disagree ;-)

Reply:
I was a bit worried that the php block was getting too long, but something like this is definitely more readable

```suggestion
                $iris_url = urlencode("https://iris-iam.stfc.ac.uk/");
                $iam_redirect = "https://" . $hostname . "/portal/redirect_uri?iss=" . $iris_url . "&" . $iam_target;
```